### ISSUE TITLE
[beta] webui: approve/disregard buttons have wrong order

### ISSUE THREAD

**Issue by jengelh** [openSUSE/open-build-service#7255]
Observed
============
The Decline button is on the left hand-side, which suggests it is the primary action.
![buttons](https://user-images.githubusercontent.com/8861948/54754472-e699a880-4be3-11e9-89b6-52cefcbde729.png)

Expected Result
============
The Approve button should be immediately left of the Decline button.

How to Reproduce
===============
1. Go to '....' a submit request where you can render a Disregard action

Further Information
===============
Regression caused by https://github.com/openSUSE/open-build-service/pull/7233

**Comment by jengelh** [comment]
(Tabbing out of the <textarea> also leads straight to the Disapprove button, breaking the prior workflow that I could accept-with-comment by typing my comment text, <TAB>, <ENTER>.)

**Comment by coolo** [comment]
> (Tabbing out of the <textarea> also leads straight to the Disapprove button, breaking the prior workflow that I could accept-with-comment by typing my comment text, , .)

https://xkcd.com/1172/ :full_moon_with_face:

**Comment by jengelh** [comment]
> https://xkcd.com/1172/ full_moon_with_face

As silly as the GNOME HIG presenting command buttons in reverse order may seem to some, at least they got tabbing right in the reference implementation (GTK). ;-)

**Comment by bgeuken** [comment]
Thanks much for reporting these issues. Much appreciated.

The location of the buttons is intended. Most (or many) of the primary buttons in the old UI were already on the right and in buttons we now want the streamline that.
But you are right, we should handle the tabbing work better.

**Comment by bgeuken** [comment]
> But you are right, we should handle the tabbing work better.

After trying this out in the UI I am not that sure anymore. Having the tabbing go from right to left when it comes to buttons is probably unexpected for most users as well. It was at least for me.

So I would say we should leave it as is. What do you think?

**Comment by coolo** [comment]
I find the left alignment of these buttons rather strange. I can't think of any other UI that has buttons at the left side. Together with the button order, it puts the default action in the middle of the screen.

**Comment by jengelh** [comment]
I find the left alignment of the button the right thing. The mouse pointer is likely to be on the left half of the screen already, e.g. when
1. clicking the URL bar to put focus there
2. clicking the <body> or some <code> element to put focus there (setting it as the "active" HTML element makes keyboard navigation within the element available)
3. clicking the comment box to enter something
Unless one is a national of a RTL-language country, the muscle memory will favor the left side, let alone the eyes. This is why verticla ad banners on websites elsewhere are commonly found on the right-hand side.

**Comment by coolo** [comment]
I would plainly trust years of research on this matter - and even if everyone else is wrong, just do it for consistency reasons. Do you see this very form here in github?

**Comment by jengelh** [comment]
Since when is github a measure?
https://features.opensuse.org/306282 - left side
https://sourceforge.net/p/gsoap2/patches/174/ - left side
http://freshcode.club/submit/beebeep - left side
https://bugzilla.opensuse.org/show_bug.cgi?id=1127790 - messy mix of left and right. Technically speaking, there are elements on the LHS that kind of force the buttons to the right unwillingly. That also applies to GH.
(phpBB 3.2 demo page) - center

**Comment by coolo** [comment]
All your examples (that didn't require login) only have on button that you end your workflow with. But as soon as you have 2 choices, it's a different story

**Comment by jengelh** [comment]
At least GH gets the tab order right (which, looking at it with w3m, is done by placing the default action first HTML-wise, and then using CSS to reorder them).

**Comment by vpereira** [comment]
@jengelh  I think we will find examples for both solutions, the most important is to be consistent 

i.e: 

gitlab 

![Screenshot_2019-04-01_16-57-51](https://user-images.githubusercontent.com/37418/55337878-c7323380-549f-11e9-88fb-9188ac2d1403.png)

google chrome

![Screenshot_2019-04-01_16-59-23](https://user-images.githubusercontent.com/37418/55337888-ce594180-549f-11e9-8715-b0ed782c4a2b.png)

google chrome upload file

![Screenshot_2019-04-01_17-00-23](https://user-images.githubusercontent.com/37418/55337977-f5b00e80-549f-11e9-8080-dea5d664968d.png)

**Comment by jengelh** [comment]
Of course I am with you on the "consistency" thought: _No_ third-party website has a **left-aligned Cancel-OK** combo. PR #7300 would have fixed that and restored consistency at least with one half of the Internet.

**Comment by vpereira** [comment]
@jengelh I think, we all agree that to let on dialog with a different alignment doesn't make sense, right?  The issue can still be open, until we finish the whole bootstrap migration, and then we can revisit this issue. The PR#7300 IMO doesn't make sense now or in a some limited scope.

**Comment by jesus-montano** [comment]
then you do not want the buttons to change it sideways, you want to keep the buttons as they are?

**Comment by jengelh** [comment]
Yeah it's always the same nonsense. Break something, then refuse to fix it until some indefinite later time.

**Comment by jengelh** [comment]
>The issue can still be open, until we finish the whole bootstrap migration

 This thing has now been broken for a month, and you have not come up with any ready-to-merge counterproposal. This is basically admitting complete defeat.
This should be merged, because as everybody knows, software projects and their visions for complete overhauls have a tendency to drag on and sometimes even languish in oblivion.
